sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/jdbc/SpringTokenCredentialProviderContextProvider.java

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.cloud.autoconfigure.implementation.jdbc;
+
+import com.azure.spring.cloud.service.implementation.identity.impl.credential.provider.SpringTokenCredentialProvider;
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+/**
+ * Provide applicationContext for SpringTokenCredentialProvider.
+ */
+public final class SpringTokenCredentialProviderContextProvider implements ApplicationContextAware {
```

Comment:
We should have a test case for this, when this bean is injected into Spring's context, the `globalApplicationContext` will be altered to the Spring context.